src/formatDate.js

```diff
@@ -9,6 +9,72 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateParts = date.split(separatorFrom);
+  const formattedDate = [];
+
+  let oldIndexDay = 0;
+  let oldIndexMonth = 0;
+  let oldIndexYear = 0;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i][0] === 'Y') {
+      oldIndexYear = i;
+    }
+
+    if (fromFormat[i][0] === 'M') {
+      oldIndexMonth = i;
+    }
+
+    if (fromFormat[i][0] === 'D') {
+      oldIndexDay = i;
+    }
```

Comment:
The way the indexes for day, month, and year are being determined is not very efficient because it requires iterating over the `fromFormat` array multiple times. Consider combining these loops into a single loop to determine all the necessary indexes in one pass.